packages/plugins/experimental/plugin-script/src/components/Toolbar.tsx

```diff
@@ -27,20 +27,34 @@ export type ToolbarProps = {
   error?: string;
 };
 
-export const Toolbar = ({ binding, onBindingChange, deployed, onDeploy, functionUrl, error }: ToolbarProps) => {
+export const Toolbar = ({
+  binding: _binding,
+  onBindingChange,
+  deployed,
+  onDeploy,
+  functionUrl,
+  error,
+}: ToolbarProps) => {
   const { t } = useTranslation(SCRIPT_PLUGIN);
   const [pending, setPending] = useState(false);
   const [editing, setEditing] = useState(false);
+  const [binding, setBinding] = useState(_binding);
   const inputRef = useRef<HTMLInputElement>(null);
 
+  useEffect(() => {
```

Comment:
see useDefaultValue

Reply:
forgot about that one, will update

Reply:
actually not sure it handles this case because:
1. there's not a stable default value
2. want to make sure it only swaps it when not in editing mode